esbuild.config.mjs

```diff
@@ -0,0 +1,75 @@
+import * as esbuild from 'esbuild';
+import * as fs from 'fs/promises';
+import * as path from 'path';
+import { fileURLToPath } from 'url';
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+const srcDir = path.join(__dirname, 'src');
+const actionsDir = path.join(__dirname, 'actions');
+const buildDir = path.join(__dirname, 'build'); // Build output directory
+
+// Shared esbuild options
+const sharedEsbuildOptions = {
+  bundle: true,       // Bundle dependencies
+  platform: 'node',   // Target Node.js environment
+  target: 'node20',   // Target Node.js version (must match action.yml)
+  minify: true,       // Minify output code
+  sourcemap: true,    // Generate source maps for debugging
+  // Mark @actions/* packages as external, they are provided by the runner
+  external: ['@actions/core', '@actions/exec', '@actions/github'],
```

Comment:
maybe you should not mark them as externals.

not sure if `@actions/* packages as external, they are provided by the runner` is correct.

checked the build artifact in example repo, and didn't see any import for `@actions/core` or any `@actions/*` packages.
https://github.com/actions/javascript-action/blob/main/dist/index.js

so you probably should bundle them instead of marking as external.